common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java

```diff
@@ -98,267 +98,360 @@
+    static ImmutableTable.Builder<TagPrefix, Material, BlockEntry<ItemPipeBlock>> ITEM_PIPE_BLOCKS_BUILDER = ImmutableTable.builder();
 
+
+    //////////////////////////////////////
+    //*****    Reference Tables    *****//
+    //////////////////////////////////////
     public static Table<TagPrefix, Material, BlockEntry<? extends MaterialBlock>> MATERIAL_BLOCKS;
+    public static Map<Material, BlockEntry<SurfaceRockBlock>> SURFACE_ROCK_BLOCKS;
+    public static Table<TagPrefix, Material, BlockEntry<CableBlock>> CABLE_BLOCKS;
+    public static Table<TagPrefix, Material, BlockEntry<FluidPipeBlock>> FLUID_PIPE_BLOCKS;
+    public static Table<TagPrefix, Material, BlockEntry<ItemPipeBlock>> ITEM_PIPE_BLOCKS;
+    public static final BlockEntry<LaserPipeBlock>[] LASER_PIPES = new BlockEntry[LaserPipeType.values().length];
+
+
+    //////////////////////////////////////
+    //*****    Procedural Blocks   *****//
+    //////////////////////////////////////
+
+    // Compressed Blocks
     public static void generateMaterialBlocks() {
-        REGISTRATE.creativeModeTab(() -> GTCreativeModeTabs.MATERIAL_BLOCK);
-        ImmutableTable.Builder<TagPrefix, Material, BlockEntry<? extends MaterialBlock>> builder = ImmutableTable.builder();
+        GTCEu.LOGGER.info("Generating Compressed Blocks...");
         for (Material material : GTRegistries.MATERIALS) {
-            // Compressed Block
-            if ((material.hasProperty(PropertyKey.INGOT) || material.hasProperty(PropertyKey.GEM) || material.hasFlag(FORCE_GENERATE_BLOCK))
-                    && !TagPrefix.block.isIgnored(material)) {
-                var entry = REGISTRATE.block("%s_block".formatted(material.getName()), properties -> new MaterialBlock(properties.noLootTable(), TagPrefix.block, material))
-                        .initialProperties(() -> Blocks.IRON_BLOCK)
-                        .transform(unificationBlock(TagPrefix.block, material))
-                        .addLayer(() -> RenderType::translucent)
-                        .setData(ProviderType.BLOCKSTATE, NonNullBiConsumer.noop())
-                        .setData(ProviderType.LANG, NonNullBiConsumer.noop())
-                        .setData(ProviderType.LOOT, NonNullBiConsumer.noop())
-                        .color(() -> MaterialBlock::tintedColor)
-                        .item(MaterialBlockItem::create)
-                        .onRegister(MaterialBlockItem::onRegister)
-                        .model(NonNullBiConsumer.noop())
-                        .color(() -> MaterialBlockItem::tintColor)
-                        .build()
-                        .register();
-                builder.put(TagPrefix.block, material, entry);
+            if (allowMaterialBlock(material)) {
+                registerMaterialBlock(material);
             }
+        }
+        GTCEu.LOGGER.info("Generating Compressed Blocks... Complete!");
+    }
+    public static boolean allowMaterialBlock(Material material) {
```

Comment:
The `allow...()` and `register...()` methods should be private

Reply:
Resolved in next push